tracing/tracer.go

```diff
@@ -0,0 +1,76 @@
+package tracing
+
+import (
+	"io"
+	"strings"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/sirupsen/logrus"
+	jeagerConf "github.com/uber/jaeger-client-go/config"
+)
+
+type Tracer struct {
+	ServiceName  string
+	Provider     string
+	Logger       *logrus.Logger
+	JaegerConfig *JaegerConfig
+
+	tracer opentracing.Tracer
+	closer io.Closer
+}
+
+type JaegerConfig struct {
+	LocalAgentHostPort string
+	SamplerType        string
+	SamplerValue       float64
+	SamplerServerUrl   string
+}
+
+func (t *Tracer) Setup() {
+	switch strings.ToLower(t.Provider) {
+	case "jaeger":
+		jc := jeagerConf.Configuration{
+			Sampler: &jeagerConf.SamplerConfig{
+				SamplingServerURL: t.JaegerConfig.SamplerServerUrl,
+				Type:              t.JaegerConfig.SamplerType,
+				Param:             t.JaegerConfig.SamplerValue,
+			},
+			Reporter: &jeagerConf.ReporterConfig{
+				LocalAgentHostPort: t.JaegerConfig.LocalAgentHostPort,
+			},
+		}
+
+		closer, err := jc.InitGlobalTracer(
+			t.ServiceName,
+		)
+
+		if err != nil {
+			t.Logger.Errorf("Could not initialize jaeger tracer: %s", err.Error())
```

Comment:
Should this be handled "silently", aka no service crash? I'm not sure if it should.